src/frontend/komponenter/Behandling/VedtakOgBeregning/VedtakOgBeregning.tsx

```diff
@@ -0,0 +1,199 @@
+import { AxiosRequestConfig } from 'axios';
+import DataViewer from '../../Felleskomponenter/DataViewer/DataViewer';
+import { Select, Textarea } from 'nav-frontend-skjema';
+import { formaterNullableIsoDato, formaterNullableMånedÅr } from '../../../utils/formatter';
+import { Søknadsgrunnlag } from '../../Felleskomponenter/Visning/DataGrunnlagIkoner';
+import { Element, Normaltekst } from 'nav-frontend-typografi';
+import { ISøknadData } from '../../../typer/beregningssøknadsdata';
+import { useApp } from '../../../context/AppContext';
+import { useDataHenter } from '../../../hooks/felles/useDataHenter';
+import { AlertStripeAdvarsel, AlertStripeFeil } from 'nav-frontend-alertstriper';
+import { useHistory } from 'react-router-dom';
+import { ModalAction, ModalType, useModal } from '../../../context/ModalContext';
+import { EBehandlingResultat, IVedtak } from '../../../typer/vedtak';
+
+interface Props {
+    behandlingId: string;
+}
+
+const StyledSelect = styled(Select)`
+    max-width: 200px;
+`;
+
+const StyledInntekt = styled.div`
+    padding: 2rem;
+`;
+
+const StyledFeilmelding = styled(AlertStripeFeil)`
+    margin-top: 2rem;
+`;
+
+const StyledAdvarsel = styled(AlertStripeAdvarsel)`
+    margin-top: 2rem;
+`;
+
+const VedtakOgBeregning: FC<Props> = ({ behandlingId }) => {
+    const { axiosRequest } = useApp();
+    const { modalDispatch } = useModal();
+    const history = useHistory();
+    const [resultatType, settResultatType] = useState<EBehandlingResultat>();
+    const [periodeBegrunnelse, settPeriodeBegrunnelse] = useState<string>('');
+    const [inntektBegrunnelse, settInntektBegrunnelse] = useState<string>('');
+    const [feilmelding, settFeilmelding] = useState<string>('');
+
+    const søknadDataConfig: AxiosRequestConfig = useMemo(
+        () => ({
+            method: 'GET',
+            url: `/familie-ef-sak/api/beregning/${behandlingId}/hent-soknad`,
+        }),
+        [behandlingId]
+    );
+
+    const søknadDataResponse: Ressurs<ISøknadData> = useDataHenter<ISøknadData, null>(
+        søknadDataConfig
+    );
+
+    const lagBlankett = () => {
+        resultatType &&
+            axiosRequest<string, IVedtak>({
+                method: 'POST',
+                url: `/familie-ef-sak/api/beregning/${behandlingId}/lagre-vedtak`,
+                data: {
+                    resultatType,
+                    periodeBegrunnelse,
+                    inntektBegrunnelse,
+                },
+            }).then((res: Ressurs<string>) => {
+                switch (res.status) {
+                    case RessursStatus.SUKSESS:
+                        history.push(`/behandling/${behandlingId}/blankett`);
+                        break;
+                    case RessursStatus.HENTER:
+                    case RessursStatus.IKKE_HENTET:
+                        break;
+                    default:
+                        settFeilmelding(res.frontendFeilmelding);
+                }
```

Comment:
Dette "pattern" går igjen noen ganger... har vi, eller bør vi ha en generisk løsning for dette? Noe lignende dataviewer, men som "gjør noe" istedenfor å vise noe.

Reply:
```
pakkUtRessurs((ressurs: Ressurs) =>{ }, (ressurs: Ressurs) =>{ })
// Første er suksess, andre er Feilet/funksjonellFeil
```

Kan det der funke?

Reply:
Eller, blir kanskje like dumt med callbacks...